### PR TITLE
AOP2-3 referral button

### PR DESCRIPTION
**Pull request by jasonfungsing**

### CODE REVIEW

File: backend/app/api/views/users.py [hunk 1]
```diff
@@ -111,7 +112,9 @@ def login():
 
         login_user(user)
 
-        return jsonify(user_info(user))
+        user_organisation = users.get_user_organisation(get_email_domain(email_address))
+
+        return jsonify(user_info(user_organisation))
```
Comment (bsigney) [inline backend/app/api/views/users.py]: update the swagger doc with new field

File: frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  componentDidMount() {
+    this.props.loadSupplierData(this.props.id)
+    console.log('ReferralBuilder props: ', this.props)
+  }
+
+  retrieveRegionName(regionCode, regionData) {
+    let regionName
+    let subRegionName
+    for (const region of regionData.regions) {
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js]: when iterating on arrays don't use for.  Use `forEach`:

```
regionData.regions.forEach((region) => {
  // DO something
});
```

Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js]: can also write this section by using `find`:

```
const { regions } = regionData;
const foundRegion = regions.find((region) => {
  const { subRegions } = region;
  const foundSubRegion = subRegions.find((subRegion) => {
    return subRegion.id = regionCode;
  });
  
  if (foundSubRegion) {
    subRegionName = foundSubRegion.name;
  }

  return foundSubRegion;
});

if (foundRegion) {
  regionName = foundRegion.name;
}
```

File: backend/app/api/views/users.py (hunk 1 above)
Comment (jasonfungsing) [reply backend/app/api/views/users.py]: done

File: frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  goBack = (props) => {
+    this.props.history.goBack()
+  }
+
+  render() {
+    const { supplierData, regionCode, regionsData, price, organisation } = this.props
+
+    return (
+      <div>
+        {supplierData && regionCode && regionsData && price && organisation
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js]: for ease of code readability:

- put this check in a readable constant: e.g. `showReferralInfo`
- put the subsequent html into another function, so code reads:

```
const showReferralInfo = supplierData && regionCode && regionsData && price && organisation;
if (showReferralInfo) {
  {this.renderReferralInfo()}
}
else {
  <LoadingIndicatorFullPage/>
}

File: frontend/apps/orams/components/ResultsTable/ResultsTable.js [hunk 4]
```diff
@@ -30,7 +30,13 @@ class ResultsTable extends Component {
     }
   }
 
+  onClickReferral(supplierCode, price){
+    this.props.setSelectedSupplierPrice(price)
+    this.props.history.push(`/referral-builder/${supplierCode}`)
+  }
+
   render(props) {
+    console.log('ResultsTable:', this.props)
```
Comment (bsigney) [inline frontend/apps/orams/components/ResultsTable/ResultsTable.js]: remove this

Comment (jasonfungsing) [reply frontend/apps/orams/components/ResultsTable/ResultsTable.js]: done

File: frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js (hunk 3 above)
Comment (jasonfungsing) [reply frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js]: done

**Pull request merged**